server/routes/auth.ts

```diff
@@ -56,25 +54,12 @@ authRoutes.get("/validatehost/:nonce", (request, response) => {
 	response.send(crypto.createHmac("sha256", config.secrets.session).update(nonce).digest().toString("hex"));
 });
 
-authRoutes.all("/logout", (request, response) => {
+authRoutes.all("/logout", async (request, response) => {
 	let user = request.user as IUser | undefined;
 	if (user) {
 		let groundTruthURL = new URL(config.secrets.groundTruth.url);
-		let requester = groundTruthURL.protocol === "http:" ? http : https;
-		let req = requester.request({
-			host: groundTruthURL.hostname,
-			path: "/api/user/logout",
-			method: "POST",
-			headers: {
-				"Authorization": `Bearer ${user.token}`
-			}
-		});
-
-		req.on('error', (e) => {
-			console.error("Unable to logout: %s", e);
-		});
-
-		req.end();
+		// Invalidates token and logs user out of Ground Truth too
+		await GroundTruthStrategy.apiRequest("POST", new URL("/api/user/logout", groundTruthURL).toString(), user.token || "");
```

Comment:
I implemented it this way because typically people want to sign out of registration + Ground Truth (i.e. to switch accounts or for security or whatever) when they click "Log out" from within registration and combining it into a single, automatic step seemed easier. I don't really see a use case for wanting to log out of registration but not Ground Truth